apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/service/ConsumerService.java

```diff
@@ -186,6 +188,29 @@ public List<ConsumerRole> assignNamespaceRoleToConsumer(String token, String app
     return Arrays.asList(createdModifyConsumerRole, createdReleaseConsumerRole);
   }
 
+  @Transactional
+  public ConsumerRole assignCreateApplicationRoleToConsumer(String token) {
+    Long consumerId = getConsumerIdByToken(token);
+    if (consumerId == null) {
+      throw new BadRequestException("Token is Illegal");
+    }
+    Role createAppRole = rolePermissionService.findRoleByRoleName(CREATE_APPLICATION_ROLE_NAME);
```

Comment:
Use admin user's role here.

Permission:
* id 1
* PermissionType CreateApplication
* TargetId SystemRole

Role:
* id 1
* RoleName CreateApplication+SystemRole

---

Should use another Permission TargetId? For example, use `OpenapiRole` instead of `SystemRole`

Reply:
I believe it's acceptable to reuse the existing roles and permission IDs, as other open APIs, such as those for creating clusters or environments, are also employing this practice. Alternatively, do you have any specific factors to take into account on your end?

Reply:
I'm fear that `SystemRole` was used in another scene which is dangerous for openapi.

If `SystemRole` is ok for openapi, that will be great.

Reply:
I think it's ok.